### PR TITLE
#166502105 Display loader while events are still loading

### PR DESCRIPTION
**Pull request by efalayi**
#### What Does This PR Do?
This PR displays a spinner while event is being fetched on `Events page`

#### Description Of Task To Be Completed
- update Spinner component
- update events page css styles
- render spinner when events are being fetched
- fix issue with redux-persist
- set initial events state to empty object
- fix eslint errors

#### Any Background Context You Want To Provide?
N/A

#### How can this be manually tested?
- pull branch
- start app and navigate to `/events`
- A spinner should be displayed before events are rendered.

#### What are the relevant pivotal tracker stories?
[#166502105](https://www.pivotaltracker.com/story/show/166502105)

#### Screenshot(s)
events-loader

### CODE REVIEW

File: client/pages/Event/EventsPage.jsx [hunk 1]
```diff
@@ -28,38 +28,50 @@ class EventsPage extends React.Component {
...
+   * React Lifecycle hook
+   * @memberof EventsPage
+   * @returns {null}
+  */
   componentDidMount() {
     const { eventStartDate } = this.state;
+
+    this.setState({ isLoadingEvents: true });
     this.getEvents({ startDate: eventStartDate });
-    this.getCategories({
```
Comment (Alameen688) [inline client/pages/Event/EventsPage.jsx:44]: Thanks for removing the duplicate  `getCategories`  request.

File: client/store/configureStore.js [hunk 2]
```diff
@@ -7,7 +7,9 @@ import rootReducer from '../reducers/index';
 import saveTokenMiddleware from '../middleware/auth';
 
 // blacklist ui state so they're not persisted
-const config = { key: 'root', storage, blacklist: ['uiReducers'] };
+const config = {
+  key: 'root', storage, blacklist: ['uiReducers', 'events'],
```
Comment (Alameen688) [inline client/store/configureStore.js:11]: Can we remove `events` from the blacklist?

File: client/store/configureStore.js [hunk 3]
```diff
@@ -22,12 +24,16 @@ const configureStore = (initialState) => {
     reducers,
     compose(
       applyMiddleware(...middleware),
+      // eslint-disable-next-line no-underscore-dangle
       window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : f => f,
     )
   );
   const persistor = persistStore(store);
+  window.persistor = persistor;
```
Comment (Alameen688) [inline client/store/configureStore.js]: I think we can do without this on the window object.

**Review by Alameen688 — COMMENTED** [review]
Great work @efalayi 🎉. This implementation works well. I left a few comments, kindly help look into them. Thank you

File: client/store/configureStore.js (hunk 2 above)
Comment (efalayi) [reply client/store/configureStore.js:11]: @Alameen688 removing `events` from the blacklist persists events in the store. This means events would always be defined. Hence setting a boolean check would not work.

Comment (Alameen688) [reply client/store/configureStore.js:11]: It's okay then.

**Pull request merged**